arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Builders.kt

```diff
@@ -136,83 +139,98 @@ public inline fun <Error, A> iorNel(noinline combineError: (NonEmptyList<Error>,
 }
 
 /**
- * Implementation of [Raise] used by `ignoreErrors`.
- * You should never use this directly.
+ * Runs a computation [block] using [Raise], and ignore its outcome.
+ *
+ * This function re-throws any exceptions thrown within the [Raise] block.
+ *
+ * Read more about running a [Raise] computation in the
+ * [Arrow docs](https://arrow-kt.io/learn/typed-errors/working-with-typed-errors/#running-and-inspecting-results).
  */
-public class IgnoreErrorsRaise<N>(
-  private val raise: Raise<N>,
-  private val error: () -> N
-) : Raise<Any?> {
+public inline fun impure(block: SingletonRaise<Unit>.() -> Unit) {
+  contract { callsInPlace(block, InvocationKind.AT_MOST_ONCE) }
+  return singleton({ }, block)
+}
+
+public class SingletonRaise<in E>(private val raise: Raise<Unit>): Raise<E> {
+  @RaiseDSL
+  public fun raise(): Nothing = raise.raise(Unit)
+
   @RaiseDSL
-  override fun raise(r: Any?): Nothing =
-    raise.raise(error())
+  override fun raise(r: E): Nothing = raise()
 
   @RaiseDSL
-  public fun ensure(value: Boolean): Unit = ensure(value) { null }
+  public fun ensure(condition: Boolean) {
```

Comment:
Maybe this can be an extension function? In fact, all methods here would ideally be extensions, but without context receivers, that can't happen yet.

Reply:
Indeed. Having them as part of the class also improves their discoverability, because no additional imports are needed to use them.

Reply:
True. Note that the corresponding functions for `Raise` are extensions though, so in fact importing `arrow.core.raise.ensure` would import both the `Raise<E>` version and the `Raise<Unit>` version, if there was to be one